crates/router/src/connector/boku/transformers.rs

```diff
@@ -1,107 +1,263 @@
+}
+
+#[derive(Debug, Clone, Serialize)]
+pub enum BokuPaymentType {
+    Dana,
+    Momo,
+    Gcash,
+    GoPay,
+    Kakaopay,
+}
+
+impl fmt::Display for BokuPaymentType {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            Self::Dana => write!(f, "Dana"),
+            Self::Momo => write!(f, "Momo"),
+            Self::Gcash => write!(f, "Gcash"),
+            Self::GoPay => write!(f, "GoPay"),
+            Self::Kakaopay => write!(f, "Kakaopay"),
+        }
+    }
+}
+
+#[derive(Debug, Clone, Serialize)]
+pub enum BokuChargeType {
+    Hosted,
+}
+
+impl fmt::Display for BokuChargeType {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            Self::Hosted => write!(f, "hosted"),
+        }
+    }
 }
 
-#[derive(Default, Debug, Serialize, Eq, PartialEq)]
-pub struct BokuCard {
-    name: Secret<String>,
-    number: cards::CardNumber,
-    expiry_month: Secret<String>,
-    expiry_year: Secret<String>,
-    cvc: Secret<String>,
-    complete: bool,
+#[derive(Debug, Clone, Serialize)]
+#[serde(rename_all = "kebab-case")]
+struct BokuHostedData {
+    forward_url: String,
 }
 
 impl TryFrom<&types::PaymentsAuthorizeRouterData> for BokuPaymentsRequest {
     type Error = error_stack::Report<errors::ConnectorError>;
     fn try_from(item: &types::PaymentsAuthorizeRouterData) -> Result<Self, Self::Error> {
         match item.request.payment_method_data.clone() {
-            api::PaymentMethodData::Card(req_card) => {
-                let card = BokuCard {
-                    name: req_card.card_holder_name,
-                    number: req_card.card_number,
-                    expiry_month: req_card.card_exp_month,
-                    expiry_year: req_card.card_exp_year,
-                    cvc: req_card.card_cvc,
-                    complete: item.request.is_auto_capture()?,
-                };
-                Ok(Self {
-                    amount: item.request.amount,
-                    card,
-                })
+            api_models::payments::PaymentMethodData::Wallet(wallet_data) => {
+                Self::try_from((item, &wallet_data))
             }
-            _ => Err(errors::ConnectorError::NotImplemented("Payment methods".to_string()).into()),
+            _ => Err(errors::ConnectorError::NotSupported {
+                message: format!("{:?}", item.request.payment_method_type),
+                connector: "Adyen",
+                payment_experience: api_models::enums::PaymentExperience::RedirectToUrl.to_string(),
+            })?,
+        }
+    }
+}
+
+impl TryFrom<(&types::PaymentsAuthorizeRouterData, &api::WalletData)> for BokuPaymentsRequest {
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(
+        value: (&types::PaymentsAuthorizeRouterData, &api::WalletData),
+    ) -> Result<Self, Self::Error> {
+        let (item, wallet_data) = value;
+        let country = match get_country_code(item) {
```

Comment:
Please use the utils for getting country

Reply:
Done👍